### PR TITLE
KYLIN-4167 Phase2

### PR DESCRIPTION
**Pull request by hit-lacus**
- Fix bugs when cubing job been dropped
- Drop former segment info when submit new one
- Add new logfile for streaming coordinator more log to let diagnostic easier.

### CODE REVIEW

File: stream-coordinator/src/main/java/org/apache/kylin/stream/coordinator/assign/AssignmentsCache.java [hunk 1]
```diff
@@ -30,54 +33,70 @@
...
 import org.apache.kylin.stream.core.model.ReplicaSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+/**
+ * Introduce AssignmentsCache to reduce pressure of
+ * @see StreamMetadataStore .
+ */
 public class AssignmentsCache {
-    private static volatile AssignmentsCache instance = new AssignmentsCache();
```
Comment (nichunen) [inline stream-coordinator/src/main/java/org/apache/kylin/stream/coordinator/assign/AssignmentsCache.java]: why remove "volatile"

Comment (hit-lacus) [reply stream-coordinator/src/main/java/org/apache/kylin/stream/coordinator/assign/AssignmentsCache.java]: It is a `Singleton`.

File: stream-source-kafka/src/main/java/org/apache/kylin/stream/source/kafka/TimedJsonStreamParser.java [hunk 2]
```diff
@@ -144,7 +149,11 @@ public StreamingMessage parse(ConsumerRecord<byte[], byte[]> record) {
                 String columnName = column.getName();
                 TimeDerivedColumnType columnType = TimeDerivedColumnType.getTimeDerivedColumnType(columnName);
                 if (columnType != null) {
-                    result.add(String.valueOf(columnType.normalize(t)));
+                    if (timeZoneOffset > 0 && TimeDerivedColumnType.isTimeDerivedColumnAboveDayLevel(columnName)) {
+                        result.add(String.valueOf(columnType.normalize(t + timeZoneOffset)));
```
Comment (hit-lacus) [inline stream-source-kafka/src/main/java/org/apache/kylin/stream/source/kafka/TimedJsonStreamParser.java]: Force advance event timestamp for derived column which above DAY level.

File: stream-coordinator/src/main/java/org/apache/kylin/stream/coordinator/assign/DefaultAssigner.java [hunk 3]
```diff
@@ -33,12 +33,16 @@
...
 import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Default implementation for Assigner, assign according to the consumer task number for each cube
  *
  */
 public class DefaultAssigner implements Assigner {
+    private static final Logger logger = LoggerFactory.getLogger(Assigner.class);
```
[the remark this answers is not in the document]
Comment (hit-lacus) [reply stream-coordinator/src/main/java/org/apache/kylin/stream/coordinator/assign/DefaultAssigner.java]: Fixed.

File: core-common/src/main/resources/kylin-defaults.properties [hunk 4]
```diff
@@ -368,4 +368,37 @@ kylin.engine.spark-conf-mergedict.spark.memory.fraction=0.2
...
+#kylin.engine.livy-conf.livy-arr.jars=hdfs:///path-to-hadoop-dependency-jar
+
+
+### Realtime OLAP ###
+
+# Where should local segment cache located, for absolute path, the real path will be ${KYLIN_HOME}/${kylin.stream.index.path}
+kylin.stream.index.path=stream_index
+
+# The timezone for Derived Time Column, please check KYLIN-4010
+kylin.stream.event.timezone=GMT+8
```
[the remark this answers is not in the document]
Comment (hit-lacus) [reply core-common/src/main/resources/kylin-defaults.properties]: Fixed.

**Pull request merged**